tests/foreman/cli/test_user.py

```diff
@@ -1421,27 +1369,21 @@ def test_negative_delete_user_1(self, opts):
         @Assert: User is not deleted
 
         """
-        login = opts.get('login')
-        if login is None:
-            login = make_user(opts)['login']
-
-        user = UserObj
-        user.katello_user = login
-        user.katello_passwd = opts.get('password', gen_alphanumeric())
-        with self.assertRaises(CLIReturnCodeError):
-            user.delete({'login': 'admin'})
-        result = UserObj.exists(search=('login', 'admin'))
-        self.assertTrue(result)
-
-    @data(
-        {'login': gen_string("alpha")},
-        {'login': gen_string("alphanumeric")},
-        {'login': gen_string("numeric")},
-        {'login': gen_string("latin1")},
-        {'login': gen_string("utf8")},
-        {'login': gen_string("alphanumeric", 100)}
-    )
-    def test_list_user_1(self, test_data):
+        for opts in ({'admin': 'true'},
+                     {'login': 'admin', 'password': 'changeme'}):
+            with self.subTest(opts):
+                login = opts.get('login')
+                if login is None:
+                    login = make_user(opts)['login']
+                user = UserObj
```

Comment:
NACK nitpick since this test resets katello_user and katello_passwd from the base class.  If the general `user` is used, it will mess up the credentials of all tests.